Tools/clinic/libclinic/dsl_parser.py

```diff
@@ -1102,6 +1102,8 @@ def bad_node(self, node: ast.AST) -> None:
                     fail("A 'defining_class' parameter cannot have a default value.")
                 if self.group:
                     fail("A 'defining_class' parameter cannot be in an optional group.")
+                if self.block.signatures[-1].cls is None:
+                    fail("A 'defining_class' parameter can be specified inside a class method.")
```

Comment:
I don't understand the error message. Do you mean "can only be defined"?

Maybe say "cannot be defined at module level: can only be defined on a class method"? Or something like that.